packages/validator/src/services/doppelgangerService.ts

```diff
@@ -54,16 +58,41 @@ export class DoppelgangerService {
     this.logger.info("Doppelganger protection enabled", {detectionEpochs: DEFAULT_REMAINING_DETECTION_EPOCHS});
   }
 
-  registerValidator(pubkeyHex: PubkeyHex): void {
+  async registerValidator(pubkeyHex: PubkeyHex): Promise<void> {
     const {currentEpoch} = this.clock;
     // Disable doppelganger protection when the validator was initialized before genesis.
     // There's no activity before genesis, so doppelganger is pointless.
-    const remainingEpochs = currentEpoch <= 0 ? 0 : DEFAULT_REMAINING_DETECTION_EPOCHS;
+    let remainingEpochs = currentEpoch <= 0 ? REMAINING_EPOCHS_IF_SKIPPED : DEFAULT_REMAINING_DETECTION_EPOCHS;
     const nextEpochToCheck = currentEpoch + 1;
 
     // Log here to alert that validation won't be active until remainingEpochs == 0
     if (remainingEpochs > 0) {
-      this.logger.info("Registered validator for doppelganger", {remainingEpochs, nextEpochToCheck, pubkeyHex});
+      const previousEpoch = currentEpoch - 1;
+      const attestedInPreviousEpoch = await this.slashingProtection.hasAttestedInEpoch(
+        fromHexString(pubkeyHex),
+        previousEpoch
+      );
+
+      if (attestedInPreviousEpoch) {
+        // It is safe to skip doppelganger detection
+        // https://github.com/ChainSafe/lodestar/issues/5856
+        remainingEpochs = REMAINING_EPOCHS_IF_SKIPPED;
+        this.logger.info("Doppelganger detection skipped, attestation from previous epoch exists in database", {
```

Comment:
Log messages to consider

- Doppelganger detection skipped. Restart detected as attestation from previous epoch exists in database
- Doppelganger detection skipped. Restart detected as previous epoch attestation exists in database
- Doppelganger detection skipped. Likely restart detected. Attestation from previous epoch exists in database
- Doppelganger detection skipped.  Assuming restart condition because attestation from previous epoch exists in database

Reply:
Opted to remove technical details on how we detected a restart from the logs. It should not be relevant for end users.